src/libs/actions/Report.ts

```diff
@@ -2488,6 +2488,128 @@ function getReportPrivateNote(reportID: string) {
     API.read('GetReportPrivateNote', parameters, {optimisticData, successData, failureData});
 }
 
+/**
+ * Add up to two report actions to a report. This method can be called for the following situations:
+ *
+ * - Adding one comment
+ * - Adding one attachment
+ * - Add both a comment and attachment simultaneously
+ */
+function completeEngagementModal(text: string, choice: string) {
+    const commandName = 'CompleteEngagementModal';
+    const conciergeAccountID = PersonalDetailsUtils.getAccountIDsByLogins([CONST.EMAIL.CONCIERGE])[0];
+    const reportComment = ReportUtils.buildOptimisticAddCommentReportAction(text, undefined, conciergeAccountID);
+    const reportCommentAction: Partial<ReportAction> = reportComment.reportAction;
+    const lastComment = reportCommentAction?.message?.[0];
+    const lastCommentText = ReportUtils.formatReportLastMessageText(lastComment?.text ?? '');
+    const reportCommentText = reportComment.commentText;
+    const currentTime = DateUtils.getDBTime();
+
+    const optimisticReport: Partial<Report> = {
+        lastVisibleActionCreated: currentTime,
+        lastMessageTranslationKey: lastComment?.translationKey ?? '',
+        lastMessageText: lastCommentText,
+        lastMessageHtml: lastCommentText,
+        lastActorAccountID: currentUserAccountID,
+        lastReadTime: currentTime,
+    };
+
+    const conciergeChatReport = ReportUtils.getChatByParticipants([conciergeAccountID]);
+    conciergeChatReportID = conciergeChatReport?.reportID;
+
+    const report = ReportUtils.getReport(conciergeChatReportID);
+
+    if (isNotEmptyObject(report) && ReportUtils.getReportNotificationPreference(report) === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN) {
+        optimisticReport.notificationPreference = CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS;
+    }
+
+    // Optimistically add the new actions to the store before waiting to save them to the server
+    const optimisticReportActions: OnyxCollection<NullishDeep<ReportAction>> = {};
+    if (reportCommentAction?.reportActionID) {
+        optimisticReportActions[reportCommentAction.reportActionID] = reportCommentAction;
+    }
```

Comment:
Should we abstract or remove? Looks like repetition from `addActions()`.

Reply:
Which part? Just the lines you highlighted here? If so I don't think that's necessary for 4 lines of code that isn't super repeatable


The optimistic data below is slightly different in this function so we can't abstract that part

Reply:
>If so I don't think that's necessary for 4 lines of code that isn't super repeatable

It looks like you copy and pasted the `addActions()` method and brought in some stuff that we might not actually need. If we don't need this line (and potentially other lines) - I think it would be better to remove them than repeat them.